GeeksCoreLibrary/Modules/PostalServices/PostNL/Controllers/PostNLController.cs

```diff
@@ -0,0 +1,231 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.Extensions;
+using GeeksCoreLibrary.Core.Interfaces;
+using GeeksCoreLibrary.Core.Models;
+using GeeksCoreLibrary.Modules.Communication.Interfaces;
+using GeeksCoreLibrary.Modules.GclReplacements.Interfaces;
+using GeeksCoreLibrary.Modules.PostalServices.PostNL.Interfaces;
+using GeeksCoreLibrary.Modules.PostalServices.PostNL.Models;
+using GeeksCoreLibrary.Modules.PostalServices.PostNL.Services;
+
+namespace GeeksCoreLibrary.Modules.PostalServices.PostNL.Controllers
+{
+    [Route("api/post-nl")]
+    public class PostNLController : Controller
+    {
+        private readonly IPostNLService postNlService;
+        private readonly IWiserItemsService wiserItemsService;
+        private readonly ICommunicationsService communicationService;
+        private readonly IStringReplacementsService stringReplacementsService;
+
+        public PostNLController(IPostNLService postNLService, IWiserItemsService wiserItemsService, ICommunicationsService communicationsService, IStringReplacementsService stringReplacementsService)
+        {
+            this.postNlService = postNLService;
+            this.wiserItemsService = wiserItemsService;
+            this.communicationService = communicationsService;
+            this.stringReplacementsService = stringReplacementsService;
+        }
+
+        private async Task<string> GetCountryCode(WiserItemModel orderDetails, string entityName)
+        {
+            var orderCountryCode = orderDetails.GetDetailValue(entityName);
+            string countryCode;
+            ulong result;
+
+            if (UInt64.TryParse(orderCountryCode, out result))
+            {
+                ulong countryId = UInt64.Parse(orderCountryCode);
+                var countryItem = await wiserItemsService.GetItemDetailsAsync(countryId);
+
+                countryCode = countryItem.GetDetailValue("name_short")?.ToUpper();
+            }
+            else
+            {
+                countryCode = orderCountryCode;
+            }
+
+            return countryCode;
+        }
+
+        [HttpGet, Route("shipping-label")]
+        public async Task<IActionResult> GenerateShippingLabel([FromQuery] string encryptedOrderIds)
+        {
+            if (String.IsNullOrWhiteSpace(encryptedOrderIds))
+            {
+                return BadRequest("Geen order ID meegegeven.");
+            }
+
+            var result = new List<string>();
+
+            foreach (var encryptedId in encryptedOrderIds.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
+            {
+                var orderId = encryptedId.DecryptWithAes(withDateTime: true, minutesValidOverride: 30);
+                var postNLDetails = await wiserItemsService.GetItemDetailsAsync(1118);
```

Comment:
Hier staat een hard-coded ID, dat kan nooit goed zijn. Graag dit nakijken en verbeteren.

Reply:
Daarnaast staat er _veel_ te veel code in deze functie. Een controller mag niets anders doen dan het verwerken van de input, aanroepen van services en zorgen voor de juiste output. Deze code moet allemaal in een service staan en dan moet je die service aanroepen in de controller.

Het kan best zijn dat dit in het originele project waar je dit vandaan had zo stond, maar in de GCL moet dit gewoon goed zijn, dus dat is geen excuus. Dan had je het in de GCL meteen moeten verbeteren